### PR TITLE
Update Changelog

### PR DESCRIPTION
**Pull request by windymilla**
First draft at Changelog for version 1.1.0

### CODE REVIEW

**Comment by windymilla** [comment]
This is a first draft. I wasn't sure where to draw the line on major/other features or even one or two bug fixes. Please feel free to suggest changes in which section, or the order of the bullet points, as well as wording.

**Comment by cpeel** [comment]
Creating good changelogs is an art and I am no artisan. What we do for the [dproofreaders releases](https://github.com/DistributedProofreaders/dproofreaders/releases) is only focus on the most important things and let users (who are arguably more developer-centric than GG users) refer back to the git log if they want the low-level details.

I would probably drop a lot of the low-level details that few, if any, people would care about -- like the OK/Close/Cancel button bit, warning users if their perl doesn't support unicode, etc or small bugfixes that should have Just Worked to begin with and only a very few people ran into -- and order entries in the most surprising / notable order first. Possibly grouped by major themes with a "catch all" for everything else. What that order should be I can't really say not being a GG user, so you might have them in this order already and I wouldn't know.

Some things that I think would be surprising / notable would include (but not be limited to):
- New Features
  - poetry HTML generation now matches Best Practices document
  - changes to HTML illustration code generated by Auto Illus Search or
    Markup Image
     - add id to the fig div based on the image filename
     - use CSS classes instead of styles on image divs
     - width of image may be specified in percent or em
     - calculate maximum percentage width for image to fit in portrait or
       landscape windows
  - ...
- Better UTF-8 support
  - bookloupe tool added as replacement for gutcheck by default
  - HTML generation defaults to keep utf-8 chars and use CSS for blockquote
  - Files now treated as utf-8 by default, rather than varying treatment
     depending on contents
- Major packaging changes
  - Use Strawberry Perl rather than old bundled version
  - include latest version of EBookMaker, runnable from HTML menu
  - Jeebies tool updated to latest version (0.15a - 2009)
  - instructions for installing and use on modern MacOS
- Removed features
  - remove RST and PGTEI menu entries, code and files
  - remove `EPub friendly` check

So in short, I'm a bad judge of what to include but there's my US$0.02 on some themes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,91 @@
...
+- /I...I/ or /i...i/ markup for index generation introduced
+- incorporate Tony Browne's regex and Greek patches (aka 1.0.28)
+- bookloupe tool added as replacement for gutcheck by default
+- search/replace preserves the position of page markers
+- autotable now uses CSS rather than HTML attributes
+- user can change how many replacement terms are in search/replace dialog
+- Count button (Ctrl+N) added to search/replace dialog to count how many
+  times the current search settings would find a match
+
+### Other New Features
```
Comment (srjfoo) [inline CHANGELOG.md:29]: How about `Other Changes` instead of `Other New Features`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,91 @@
...
+  included by default
+- Goto Line, Page and Page Label dialogs can now be closed with Escape keys
+  or Window Manager close button
+- operation tracking now defaults to "off" to reduce user confusion caused
+  by the "edited" flag being set
+- style Guide for developers created to improve consistency
+   - OK/Cancel/Close now used to close dialogs more consistently
+   - dialog titles made more consistent with menu options which popped them
+
+### Bug Fixes
```
Comment (srjfoo) [inline CHANGELOG.md:66]: Below, sometimes you say what was done (`word Frequency harmonics code fixed to spot single letter change`), other times you just describe the bug(`dual monitors on Macs could lead to an over-large initial window`)

**Comment by srjfoo** [comment]
> So in short, I'm a bad judge of what to include but there's my US$0.02 on some themes.

I'm in something of the same position. I made some specific comments, but I don't think I'm all that qualified to decide. I will say that I like to see mention of bug fixes when I'm looking at changelogs.

**Comment by windymilla** [comment]
Oops. A little knowledge is a dangerous thing :) I attempted to squash my 2 commits into 1 using rebase as I thought this was a small enough commit, it wouldn't really matter if it went wrong. The rebase went fine, but I then accidentally did "fetch origin" in GitHub desktop instead of a forced push in the bash shell. So ended up with 3 commits. I did another rebase to remove all three (after failing to remove the ones I didn't want due to conflicts), and then force pushed that. I'm guessing Git then said "oh, you don't have any commits, I'll close the Pull request." What I think I should have done was add my change and recommit it before the force push (or not made any of the earlier mistakes). Anyway, I pushed my new commit and reopened the PR. Let me know if you want me to close it and start again with a fresh branch/PR. The good news is that I think I have a much better understanding - my only previous attempts at this type of thing have been just typing the commands I've been given without properly knowing what they do.

**Comment by windymilla** [comment]
I'm also sending a copy of this to Charlie as suggested by @srjfoo

**Comment by cpeel** [comment]
I like the theming and the ordering, nicely done! This LGTM. I'm happy to merge this when you give me the go-ahead (presumably after you hear back from Charlie).

> Let me know if you want me to close it and start again with a fresh branch/PR.

Your commits in this branch / MR look good to me, so no need to open a new one.

**Comment by windymilla** [comment]
Thanks - Charlie is happy with it. Ready to merge please.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,72 @@
...
+## Version 1.1.0
+
+### Improved HTML generation
+- poetry HTML generation matches DP Best Practices document
+- illustration code generated by Auto Illus Search or Markup Image
+   - adds id to the fig div based on the image filename
+   - uses CSS classes instead of styles on image divs
+   - width of image may be specified in percent or em
+   - calculates max width for image to fit portrait or landscape screens
+- /I...I/ or /i...i/ markup is used to generate an index
```
Comment (cpeel) [inline CHANGELOG.md:12]: Consider putting the markup in single quotes:
```
`/I...I/` or `/i...i/`
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,72 @@
...
+- HTML/CSS for chapter headings works well for ePub formats
+- default <hr> CSS defines margins to center correctly in ePub
```
Comment (cpeel) [inline CHANGELOG.md:14]: We need to put the `<hr>` and `<a>` below this in single quotes to prevent it from being rendered.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,72 @@
...
+- uses Strawberry Perl rather than old bundled version
+- includes latest version of EBookMaker, runnable from HTML menu
+- Jeebies tool is updated to latest version (0.15a - 2009)
+- ppvimage tool is updated to match new image size guidance
+- git checkout can be used as a live release (developers/testers)
+
+### Other Changes
+- Tony Browne's regex and Greek patches (aka 1.0.28) are included
+- new Keyboard Shortcuts are included:
+   - ctrl+o - open file; ctrl+shift+s - save as...
```
Comment (cpeel) [inline CHANGELOG.md:44]: This is rather difficult to parse when rendered, perhaps:
```
   - `ctrl+o` - open file
   - `ctrl+shift+s` - save as...
   - `ctrl+j` - goto line
   - `ctrl+n` - count number of search/replace matches
   - `ctrl+w` - rewrap selection
   - `ctrl+shift+w` - block rewrap selection
   - `ctrl+m` - indent +1
   - `ctrl+shift+m` - indent -1
   - `ctrl+alt+m` - indent +4
   - `ctrl+alt+shift+m` - indent -4
   - `ctrl+e` instead of `ctrl+o` - flood fill
```
which renders as
   - `ctrl+o` - open file
   - `ctrl+shift+s` - save as...
   - `ctrl+j` - goto line
   - `ctrl+n` - count number of search/replace matches
   - `ctrl+w` - rewrap selection
   - `ctrl+shift+w` - block rewrap selection
   - `ctrl+m` - indent +1
   - `ctrl+shift+m` - indent -1
   - `ctrl+alt+m` - indent +4
   - `ctrl+alt+shift+m` - indent -4
   - `ctrl+e` instead of `ctrl+o` - flood fill

I'm not too worried about the vertical space here.

**Comment by windymilla** [comment]
Thanks for pointing those out - hopefully that's all OK now

**Comment by windymilla** [comment]
Thanks

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,77 @@
...
+- git checkout can be used as a live release (developers/testers)
+
+### Other Changes
+- Tony Browne's regex and Greek patches (aka 1.0.28) are included
+- new Keyboard Shortcuts are included:
+   - `ctrl+o` - open file
+   - `ctrl+shift+s` - save as...
+   - `ctrl+j` - goto line
+   - `ctrl+n` - count number of search/replace matches
+   - `ctrl+w` - rewrap selection
```
Comment (srjfoo) [inline CHANGELOG.md:48]: Again, for Macs, `cmd+w` is used to close an open window/tab without quitting the application.
